### DISCUSSION TITLE
How do I check whether neovim is in operator-pending mode?

### DISCUSSION THREAD

**Discussion by sylvesterroos** [vscode-neovim/vscode-neovim#1614] [Q&A] [answered]
I have the following keybind in my `keybinds.json`, but it's preventing me from executing an operation like `dt<space>`:

```json
{
    "key": "space",
    "command": "vspacecode.space",
    "when": "editorTextFocus && neovim.mode == normal"
}
```
I would like to expand the `when` with something like `&& neovim.mode != operator-pending`, but I'm not sure which option to use, if the option exists at all.

Thanks :)

**Comment by justinmk** [comment]
Remove that vscode mapping and define the mapping in your nvim init.lua file instead.

    nnoremap <silent> <space> <cmd>lua require('vscode-neovim').call('vspacecode.space')<cr>

**Reply by sylvesterroos** [reply]
Thanks for your quick answer! The code you supplied appears to be vimscript. Do you know how I can achieve the same in lua?

I'm entirely ignorant on neovim configuration 😅

**Reply by xiyaowong** [reply]
`:h lua-guide`

**Reply by justinmk** [reply, marked as answer]
Just put it in a `vim.cmd` call.

```lua
vim.cmd[[nnoremap <silent> <space> <cmd>lua require('vscode-neovim').call('vspacecode.space')<cr>]]
```

**Comment by carschandler** [comment]
@justinmk is there a way to actually use operator-pending mode in the when condition? I have a different use case: when using Jupyter Notebooks and you start an action like `d` or `c` but decide you want to cancel it, the Stop Editing Cell keybind takes priority. I know I could use `Ctrl`+`[` instead, and I've also mapped `Shift`+`Escape` to leave notebook cells before, opening up escape to leave operator-pending mode, but it would be cool to be able to just use Escape for both of these, and the only way I can think of that is to check whether or not you're in operator-pending mode. I know `vim.api.nvim_get_mode` will indicate this in its output, so would it be possible to loop this in somehow?

**Reply by theol0403** [reply]
https://github.com/vscode-neovim/vscode-neovim/issues/1718

**Reply by carschandler** [reply]
Thank you!

**Comment by gyaaniguy** [comment]
@justinmk Your solution works great ! But for me there is a delay of 1-2 seconds before the menu shows. Any ideas ? It might be because I am using wsl2 ?

**Reply by justinmk** [reply]
you probably have a conflicting/overlapping mapping.

**Reply by gyaaniguy** [reply]
You are right ! The leader key was mapped to <space> as well. Thanks !
